### PR TITLE
DELETE Endpoint added

### LINKED ISSUE #74: Create DELETE endpoint for opportunities
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (for multi-repository projects)

# Feature Request

I can see that there are apis for creating and fetching the opportunities but no api exists for updating and deleting.  @Manvityagi @abdus Kindly look into this issue and assign it to me if it makes sense.

### PR DESCRIPTION
**Pull request by Vishal1003**
# Description

I have added the relevant changes (delete method for the opportunities). @abdus  Kindly review my changes!

Fixes #74

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

RESPONSE IMAGE -
image

### CODE REVIEW

**Review by Manvityagi — CHANGES_REQUESTED** [review]
@Vishal1003 Respective Tests for the API should also be added.

**Comment by Manvityagi** [comment]
Any updates @Vishal1003 ?

**Comment by Vishal1003** [comment]
@Manvityagi @abdus Sorry for the delay. I have done my task but the getOpportunity tests are failing (don't know why)  after I merged with the upstream branch (and I believe there is a separate issue for resolving this). 

Kindly review my changes. Thank you!

**Comment by Manvityagi** [comment]
> @Manvityagi @abdus Sorry for the delay. I have done my task but the getOpportunity tests are failing (don't know why) after I merged with the upstream branch (and I believe there is a separate issue for resolving this).
> 
> Kindly review my changes. Thank you!

Yeah, you are right. We are figuring what's wrong with it.

**Comment by Manvityagi** [comment]
@Vishal1003  Pull the latest code, the issue has been fixed

**Comment by Manvityagi** [comment]
Have you entered the right credentials in `.env` , the build fails, seems like a connection to DB could not be established.

**Comment by Vishal1003** [comment]
> Have you entered the right credentials in `.env`? The build fails, seems like a connection to DB could not be established.

@Manvityagi Yes i entered the correct credentials.. I don't know why the build is failing.. (the tests are running fine on my pc).

**Comment by Manvityagi** [comment]
> > Have you entered the right credentials in `.env`? The build fails, seems like a connection to DB could not be established.
> 
> @Manvityagi Yes i entered the correct credentials.. I don't know why the build is failing.. (the tests are running fine on my pc).

Take the pull once more, it should work now

**Comment by Vishal1003** [comment]
> > > Have you entered the right credentials in `.env`? The build fails, seems like a connection to DB could not be established.
> > 
> > 
> > @Manvityagi Yes i entered the correct credentials.. I don't know why the build is failing.. (the tests are running fine on my pc).
> 
> Take the pull once more, it should work now

Done! Kindly review my changes (for issue #74 ) Thank you

File: routes/opportunity.js [hunk 1]
```diff
@@ -236,4 +236,71 @@ router.patch('/:opportunity_id', (req, res) => {
   opportunityController.updateOpportunity(req, res);
 });
 
+/**
+ * @openapi1
```
Comment (Manvityagi) [inline routes/opportunity.js]: There is a extra `1` after @openapi, due to which this API doesn't show up in Swagger UI at http://localhost:3030/playground

File: routes/opportunity.js [hunk 2]
```diff
@@ -236,4 +236,71 @@ router.patch('/:opportunity_id', (req, res) => {
...
+ * /opportunity:
+ *   delete:
+ *     summary: Deletes an opportunity
+ *     description: Deletes an opportunity from the database based on the id provided!
+ *     parameters:
+ *      - id: type
```
Comment (Manvityagi) [inline routes/opportunity.js]:
``` 
*     parameters:
 *      - name: opportunity_id
 *        in: path
 *        schema:
 *          type: string
 *        description: mongo_id of a Particular opportunity
 *
 ```

It should be like this

File: routes/opportunity.js [hunk 3]
```diff
@@ -236,4 +236,71 @@ router.patch('/:opportunity_id', (req, res) => {
...
+ *                       description: Link for the opportunity.
+ *                       example: www.rohithmsrscholarship.com
+ *                     onlyForFemale:
+ *                       type: boolean
+ *                       description: Represents if the opportunity is only for females
+ *                       example: true
+ *
+ */
+
+router.delete('/:id', (req, res) => {
```
Comment (Manvityagi) [inline routes/opportunity.js]: change this name from `'/:id'` to `/:opportunity_id`

Comment (Manvityagi) [reply routes/opportunity.js]: Here and everywhere else

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Please do the changes as mentioned in comments

**Comment by Vishal1003** [comment]
> Please do the changes as mentioned in comments

@Manvityagi I have made all the necessary changes.

**Review by Manvityagi — APPROVED** [review]
Awesome @Vishal1003. Merging

**Pull request merged**